lib/puppet/pops/types/type_calculator.rb

```diff
@@ -452,9 +452,18 @@ def instance_of_PTupleType(t, o)
 
   def instance_of_PStructType(t, o)
     return false unless o.is_a?(Hash)
-    h = t.hashed_elements
-    # all keys must be present and have a value (even if nil/undef)
-    (o.keys - h.keys).empty? && h.all? { |k,v| instance_of(v, o[k]) }
+    match_count = 0
+    ok = t.elements.reduce(true) do |memo,e|
+      break false unless memo
+      v = o[e.name]
+      if v.nil?
+        e.type.is_a?(Types::POptionalType)
```

Comment:
This is not correct - there are other types that accept Optional, such as a Variant that includes an Optional, or Data.